### ISSUE TITLE
Unified withdrawal fee options

### ISSUE THREAD

**Issue by devsi** [ccxt/ccxt#5644]
Hi @kroitor,

This might not be relevant so please delete if it's not.

Some exchanges give you the option to `includeFee` when withdrawing. Others do not give you the option, but just do one or the other as default (and you can't change it.)

Does it make sense to have a property on the exchange that specifies which of the two it is so the withdrawals can all handle it the same way? Or is it too ambiguous?

(e.g.) ExchangeA: Assume the withdrawal fee is 10:

> - Buy 110
> - Withdraw 110
> - Error -> doesn't take in to account fee
> - Withdraw 100
> - Receive 100
> - Balance = 0. 10 taken from balance.

(e.g.) ExchangeB: Assume the withdrawal fee is 10:

> - Buy 110
> - Withdraw 110
> - Success -> Fee included
> - Receive 100
> - Balance = 0. 10 taken from withdrawal.

So could it be unified in such a way that you give CCXT the exact amount to withdraw, and internally it can convert add the fee on if the exchange supports scenario B by default?

**Comment by frosty00** [comment]
yep this is a great idea, should be an exchange specific option so that users can calculate how much to send to the exchange to withdraw their full balance. We will probably have to do some trial and error to figure out which exchanges support which `includeFee` mode.

**Comment by wukoujuey** [comment]
I can't believe how terrible people have become in the world, how can a broker call me up and make me believe that I was investing into something genuine, in a couple of months I had been fleeced out of my life savings, till this day haven't even been able to get fully back on my feet. Having taken my time to do some research, I would like to speak with other people who have been through similar experiences, together without a doubt I am sure with like minded minds we can all get our money back. I met with a genius Olivia Teams, who helped me get back on my feet again
Email oliviatrader 44 @"gmail ,com
WhatsApp  +1 (423) 248-2140
